arches/app/utils/i18n.py

```diff
@@ -0,0 +1,175 @@
+
+
+class ArchesPOFileFetcher:
+    """Gets PO files for processing"""
+
+    def get_po_file(self, lang, overwrite=False) -> polib.POFile:
+        """Gets a graph PO file for a specific language"""
+        path = settings.LOCALE_PATHS[0]
+        po_path = os.path.join(path, lang, "LC_MESSAGES")
+        pathlib.Path(po_path).mkdir(parents=True, exist_ok=True)
+        pofilepath = os.path.join(po_path, "graphs.po")
+        if overwrite or not os.path.exists(pofilepath):
+            open(pofilepath, "w").close()
+            return self.setup_file(polib.pofile(pofilepath))
+
+        for pofilepath in glob.glob(os.path.join(po_path, "graphs.po")):
+            return self.setup_file(polib.pofile(pofilepath))
+
+    def get_po_files(self, lang=None, overwrite=False) -> List[ArchesPOFile]:
+        """Get an array of PO files being loaded or written"""
+        files = []
+
+        if lang:
+            files.append((lang, self.get_po_file(lang), overwrite))
+        else:
+            for language in settings.LANGUAGES:
+                files.append(ArchesPOFile(language[0], self.get_po_file(language[0], overwrite)))
+        return files
+
+    def setup_file(self, pofile: polib.POFile) -> polib.POFile:
+        """Given a PO file, set up metadata and duplicate check property"""
+        current_time = datetime.now()
+        pofile.check_for_duplicates = True
+        pofile.metadata = {
+            "Project-Id-Version": "1.0",
+            "Report-Msgid-Bugs-To": "dev@fargeo.com",
+            "POT-Creation-Date": current_time.strftime("%Y-%m-%d %H:%M:%S%z"),
+            "PO-Revision-Date": current_time.strftime("%Y-%m-%d %H:%M:%S%z"),
+            "MIME-Version": "1.0",
+            "Content-Type": "text/plain; charset=utf-8",
+            "Content-Transfer-Encoding": "8bit",
+        }
+
+        return pofile
+
+
+class ArchesPOWriter:
+    """Writes a PO file from arches graph tables"""
+
+    def __init__(self, pofile: polib.POFile, id_language: str, target_language: str) -> None:
+        self.pofile = pofile
+        self.id_language = id_language
+        self.target_language = target_language
+
+    def populate(self):
+        """Populates a PO file with entries from several arches graph tables"""
+        self.populate_from_card_x_node_x_widget(CardXNodeXWidget.objects.all())
+        self.populate_from_cards(CardModel.objects.all())
+
+        self.pofile.save()
+
+    def populate_from_card_x_node_x_widget(self, queryset):
+        for row in queryset:
+            try:
+                i18n_properties = row.config["i18n_properties"]
+                for prop in i18n_properties:
+                    try:
+                        self.append(row.config[prop], "cards_x_nodes_x_widgets")
+                    except KeyError:
+                        pass
+                self.append(row.label, "cards_x_nodes_x_widgets")
+            except KeyError:
+                pass
```

Comment:
this still fails for me.  line 84 doesn't always get run because of a KeyError triggered on line 78.
Maybe put line 84 ahead of 78.